CMakeLists.txt

```diff
@@ -450,9 +450,9 @@ find_public_dependency(Threads REQUIRED)
 
 if(CMAKE_CXX_COMPILER_ID MATCHES Clang)
 	add_compile_options(
-		-Weverything
```

Comment:
the idea of using `-Weverything` is to have new warnings enabled, and then disable ones that aren't important or helpful. So generally I would prefer to disable warnings rather than enabling them. Is there a good reason to remove this?

Reply:
The link (https://quuxplusone.github.io/blog/2018/12/06/dont-use-weverything/) from the relevant commit quote clang developers stance on this flag. TLDR, the flag is for "It is intended strictly for Clang developers or for exploring what warnings exist" and not daily use.

Reply:
It makes me a bit nervous since it's hard to know if there are any warnings that are disabled by this change. The code honours the existing warnings, so disabling them won't be noticed by CI

Reply:
A lot of effort has been put in `libtorrent` to not have warnings and I don't recall a problem due to the use of `Weverything`. My opinion is that at this moment there is no harm in keeping it, and a lot to lose by not knowing what "interesting warnings" can come up in new clang versions.

Reply:
Well @arvidn do you really want to keep it?

Reply:
Anyway, the change is dropped.